tools/system_libs.py

```diff
@@ -1445,16 +1453,26 @@ def get_base_name(self):
 
   @classmethod
   def vary_on(cls):
-    return super().vary_on() + ['debug']
+    return super().vary_on() + ['debug', 'is_ww']
 
   @classmethod
   def get_default_variation(cls, **kwargs):
-    return super().get_default_variation(debug=settings.ASSERTIONS >= 1, **kwargs)
+    return super().get_default_variation(is_ww=settings.WASM_WORKERS, debug=settings.ASSERTIONS >= 1, **kwargs)
 
   def get_files(self):
+    files = []
+    if self.is_ww:
+      files = [
+        'library_wasm_worker.c',
+        'wasm_worker.S',
+      ]
+    else:
+      files = [
+        'library_wasm_worker_stub.c'
+      ]
     return files_in_path(
         path='system/lib/wasm_worker',
-        filenames=['library_wasm_worker.c' if self.is_ww or self.is_mt else 'library_wasm_worker_stub.c'])
+        filenames=files)
```

Comment:
And i think you revert all changes to this class except this one method?  (since libwasm_workers inherits from MTLibrary which means it already has `is_ww`, assuming you revert the change to MTLibrary too).